control-plane/pkg/reconciler/channel/v2/channelv2.go

```diff
@@ -258,7 +262,45 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, channel *messagingv1beta
 		return err
 	}
 
-	address, _ := url.Parse(fmt.Sprintf("http://%s.%s.svc.%s", channelService.Name, channelService.Namespace, network.GetClusterDomainName()))
+	var addressableStatus duckv1.AddressStatus
+	channelHttpsHost := network.GetServiceHostname(r.Env.IngressName, r.SystemNamespace)
+	channelHttpHost := network.GetServiceHostname(channelService.Name, channel.Namespace)
+	transportEncryptionFlags := feature.FromContext(ctx)
+	if transportEncryptionFlags.IsPermissiveTransportEncryption() {
+		caCerts, err := r.getCaCerts()
+		if err != nil {
+			return err
+		}
+
+		httpAddress := receiver.ChannelHTTPAddress(channelHttpHost, channelService)
+		httpsAddress := receiver.HTTPSAddress(channelHttpsHost, channelService, caCerts)
+		// Permissive mode:
+		// - status.address http address with path-based routing
+		// - status.addresses:
+		//   - https address with path-based routing
+		//   - http address with path-based routing
+		addressableStatus.Addresses = []duckv1.Addressable{httpsAddress, httpAddress}
+		addressableStatus.Address = &httpAddress
+	} else if transportEncryptionFlags.IsStrictTransportEncryption() {
+		// Strict mode: (only https addresses)
+		// - status.address https address with path-based routing
+		// - status.addresses:
+		//   - https address with path-based routing
+		caCerts, err := r.getCaCerts()
+		if err != nil {
+			return err
+		}
+
+		httpsAddress := receiver.HTTPSAddress(channelHttpsHost, channelService, caCerts)
+		addressableStatus.Addresses = []duckv1.Addressable{httpsAddress}
+		addressableStatus.Address = &httpsAddress
+	} else {
+		httpAddress := receiver.ChannelHTTPAddress(channelHttpHost, channelService)
+		addressableStatus.Address = &httpAddress
```

Comment:
Again, here it is likely worth setting `Status.Addresses`
```suggestion
		httpAddress := receiver.ChannelHTTPAddress(channelHttpHost, channelService)
		addressableStatus.Address = &httpAddress
		addressableStatus.Addresses = []duckv1.Addressable{httpAddress}
```